File: RGBWell/RGBWell/ViewController/MainViewController.m
//
//  MainViewController.m
//  RGBWell
//
//  Created by Yanice on 2016/12/3.
//  Copyright © 2016年 Yanice. All rights reserved.
//

#import "MainViewController.h"

@interface MainViewController ()

@property (weak) IBOutlet NSColorWell *ColorWell;

@property (weak) IBOutlet NSSlider *r;
@property (weak) IBOutlet NSSlider *g;
@property (weak) IBOutlet NSSlider *b;

@end

NSString *key;

@implementation MainViewController {

    CGFloat r,g,b;
}

@synthesize r=_r;
@synthesize g=_g;
@synthesize b=_b;

- (NSString *)windowNibName {

    return @"MainViewController";
}

- (void)windowDidLoad {
    [super windowDidLoad];
    
    // Implement this method to handle any initialization after your window controller's window has been loaded from its nib file.
    
    [self updateColor];
    
    [self.ColorWell addObserver:self forKeyPath:@"color" options:NSKeyValueObservingOptionNew context:&key];
}

- (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(NSDictionary<NSKeyValueChangeKey,id> *)change context:(void *)context {

    if (context == &key) {
        
        self.r.floatValue = self.ColorWell.color.redComponent;
        self.g.floatValue = self.ColorWell.color.greenComponent;
        self.b.floatValue = self.ColorWell.color.blueComponent;
    }
}

- (IBAction)redAdjust:(NSSlider *)sender {
    
    r = sender.floatValue;
    [self updateColor];
}

- (IBAction)greenAdjust:(NSSlider *)sender {

    g = sender.floatValue;
    [self updateColor];
}

- (IBAction)blueAdjust:(NSSlider *)sender {
    
    b = sender.floatValue;
    [self updateColor];
}

- (void)updateColor {

    self.ColorWell.color = [NSColor colorWithRed:r green:g blue:b alpha:1];
}

@end

File: BusyApp/BusyApp/ViewController/MainViewController.swift
//
//  MainViewController.swift
//  BusyApp
//
//  Created by Yanice on 2016/12/3.
//  Copyright © 2016年 Yanice. All rights reserved.
//

import Cocoa

class MainViewController: NSWindowController {

    @IBOutlet weak var slider: NSSlider!
    @IBOutlet weak var hints: NSTextField!
    @IBOutlet weak var show: NSButton!
    @IBOutlet weak var hidden: NSButton!
    
    var value:NSInteger!
    
    override func windowDidLoad() {
        super.windowDidLoad()

        // Implement this method to handle any initialization after your window controller's window has been loaded from its nib file.
        
        self.show.state = 1
        
        self.value = self.slider.integerValue;
        
    }
    
    @IBAction func valueChange(_ sender: NSSlider) {
        
        
        if sender.integerValue==self.value {
            return
        }else if(sender.integerValue<self.value) {
            self.hints.stringValue = "Slider Down UP!"
        }else {
            self.hints.stringValue = "Slider grow UP!"
        }
        self.value = sender.integerValue
    }
    
    @IBAction func showMarks(_ sender: NSButton) {
        self.hidden.state = 0
        self.slider.numberOfTickMarks = 10
    }
    @IBAction func hiddenMarks(_ sender: NSButton) {
        self.show.state = 0
        self.slider.numberOfTickMarks = 0
    }
    @IBAction func resetButton(_ sender: NSButton) {
        
    }
}
